components/raft_log_engine/src/engine.rs

```diff
@@ -324,6 +365,25 @@ impl RaftEngineReadOnly for RaftLogEngine {
         }
         Ok(())
     }
+
+    fn is_empty(&self) -> Result<bool> {
```

Comment:
You mean `is_empty` must returns true if there's any data file in Raft Engine? Even if logically all data is lost or compacted.

Reply:
And I also need a way to scan all the state keys from all regions.

Reply:
@BusyJay Would it be enough to get a list of raft groups (`Engine::raft_groups()`) and then fetch their states one by one?

Reply:
Yes, it works too. But how to destroy a raft group?

Reply:
Actually returning all raft group no matter if they are tombstone or not are better.

Reply:
That's not possible currently. They will be removed from memory shortly after `clean` is called.

Reply:
But tombstone key will be written by TiKV. And the state should be kept forever.

Reply:
If you intend to keep a key, then use `compact` to clean up data instead of `clean`. To record the "tombstone" state, you can delete the raft state key. Do the scan like this:
```
for rid in engine.raft_groups() {
  if let Some(state) = engine.get_message(rid, &state_key) {
    // active
  } else {
    // tombstone
  }
}
```